hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/plan/strategy/SparkSizeBasedClusteringPlanStrategy.java

```diff
@@ -91,7 +93,7 @@ protected Stream<HoodieClusteringGroup> buildClusteringGroupsForPartition(String
       totalSizeSoFar += currentSize;
     }
 
-    if (!currentGroup.isEmpty()) {
+    if (currentGroup.size() > 1 || (writeConfig.isClusteringSortEnabled() && currentGroup.size() == 1)) {
```

Comment:
Can we keep the line as is and modify the condition of line91, add the fileSlice to the currentGroup when necessary?

Reply:
Adjust the judgment position.